### PR TITLE
Fix readpixels bugs in ShaderDerivate.html

### PR DESCRIPTION
**Pull request by qkmiao**
This patch add gluPixelTransfer.readPixels which is lost when porting
C++ to Web. With this patch all tests passed in ShaderDerivate.html.

C++ reference: 
https://android.googlesource.com/platform/external/deqp/+/master/framework/opengl/gluPixelTransfer.cpp
https://android.googlesource.com/platform/external/deqp/+/master/modules/gles3/functional/es3fShaderDerivateTests.cpp#797

### CODE REVIEW

File: sdk/tests/deqp/framework/opengl/gluPixelTransfer.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+gluPixelTransfer.getTransferAlignment = function(format) {
+    var pixelSize = format.getPixelSize();
+    if (deMath.deIsPowerOfTwo32(pixelSize))
+        return Math.min(pixelSize, 8);
+    else
+        return 1;
+};
+
+gluPixelTransfer.readPixels = function(ctx, x, y, format, dst) {
+    ctx = ctx || gl;
```
Comment (zhenyao) [inline sdk/tests/deqp/framework/opengl/gluPixelTransfer.js]: Where is this gl come from? Looking at other glu files, they always pass in context and assume it's not null.

File: sdk/tests/deqp/framework/opengl/gluPixelTransfer.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    var width = dst.getWidth();
+    var height = dst.getHeight();
+    var arrayType = tcuTexture.getTypedArray(format.type);
+    var transferFormat = gluTextureUtil.getTransferFormat(format);
+    ctx.pixelStorei(gl.PACK_ALIGNMENT, gluPixelTransfer.getTransferAlignment(format));
```
Comment (zhenyao) [inline sdk/tests/deqp/framework/opengl/gluPixelTransfer.js]: context.PACK_ALIGNMENT?

File: sdk/tests/deqp/framework/opengl/gluPixelTransfer.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    var resultPixel = dst.getAccess().getDataPtr();
+    resultPixel = new arrayType(dst.getAccess().getBuffer());
+    ctx.readPixels(x, y, width, height, transferFormat.format, transferFormat.dataType, resultPixel);
+};
+
```
Comment (zhenyao) [inline sdk/tests/deqp/framework/opengl/gluPixelTransfer.js]: What about other functions in the c++ file?  Can you at least add a TODO here?

**Comment by zhenyao** [comment]
Mostly look good.  Thanks for fixing this.

**Comment by qkmiao** [comment]
Thanks @zhenyao . I updated this pull request. PTAL.

**Comment by zhenyao** [comment]
Thank you!

**Pull request merged**